### PR TITLE
Add more diagnostic records

### PR DESCRIPTION
**Pull request by Tom-Willemsen**
### Description of work

Adds extra diagnostic records

### To test

https://github.com/ISISComputingGroup/IBEX/issues/2660

### Acceptance criteria

Same things are now available in IOC as in labview

---

#### Code Review

- [ ] Is the code of an acceptable quality?
- [ ] Are the PVs named according to the [naming standards](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/PV-Naming).
- [ ] Have suitable [disable records](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Disable-records) been added?
- [ ] Have suitable [simulation records](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Record-Simulation) been added?
- [ ] Have the [finishing touches](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/IOC-Finishing-Touches) been applied?
- [ ] Is there an [emulator](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Emulating-Devices) for the device?
- [ ] Have the changes been documented in the [release notes](https://github.com/ISISComputingGroup/IBEX/wiki/ReleaseNotes_Dev). If so, do they describe the changes appropriately?
- [ ] If an OPI has been modified, does it conform to the [style guidelines](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/OPI-Creation)? There is a script called `check_opi_format.py` to help with this.

### Functional Tests

- [ ] Do changes function as described? Add comments below that describe the tests performed.
- [ ] Does the IOC respond correctly both in full and simulation mode, where it's possible to test both?
- [ ] If there are multiple _0n IOCs, do they run correctly?
- [ ] Ensure that the log files do not contain undefined macros, ie serach for `macLib: macro` full text is `macLib: macro XXXXX is undefined (expanding string XXXX)`

### Final steps

- [ ] Reviewer has updated the submodule in the main EPICS repo? See **Reviewing work for the subModules of EPICS** in the [Git workflow](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Git-workflow) page for details.
- [ ] Reviewer has moved the [release notes](https://github.com/ISISComputingGroup/IBEX/wiki/ReleaseNotes_Dev) entry for this ticket in the "Changes merged into master" section

### CODE REVIEW

File: FERMCHOP/FERMCHOP-IOC-01App/Db/fermchop.db [hunk 1]
```diff
@@ -316,6 +316,15 @@ record(ai, "$(P)AUTOZERO:2:LOWER")
 	info(archive, "VAL")
 }
 
+record(calc, "$(P)AUTOZERO:RANGECHECK")
+{
+	field(INPA, "$(P)AUTOZERO:1:UPPER CP")
+	field(INPB, "$(P)AUTOZERO:2:UPPER CP")
+	field(INPC, "$(P)AUTOZERO:1:LOWER CP")
+	field(INPD, "$(P)AUTOZERO:2:LOWER CP")
+	field(CALC, "A > 3 | B > 3 | C < -3 | D < -3")
```
Comment (DominicOram) [inline FERMCHOP/FERMCHOP-IOC-01App/Db/fermchop.db]: Can you add a comment as to where these values come from?

File: FERMCHOP/FERMCHOP-IOC-01App/Db/fermchop.db [hunk 2]
```diff
@@ -346,6 +365,13 @@ record(ai, "$(P)TEMPERATURE:MOTOR")
 	info(archive, "VAL")
 } 
 
+record(calc, "$(P)TEMPERATURE:RANGECHECK")
+{
+	field(INPA, "$(P)TEMPERATURE:MOTOR CP")
+	field(INPB, "$(P)TEMPERATURE:ELECTRONICS CP")
+	field(CALC, "A > 45 | B > 45")
```
Comment (DominicOram) [inline FERMCHOP/FERMCHOP-IOC-01App/Db/fermchop.db]: Can you add a comment as to where these values come from?

**Comment by DominicOram** [comment]
* Added a couple of comments in the code.
* Can you please add a couple of IOC system tests to cover some of this functionality

File: FERMCHOP/FERMCHOP-IOC-01App/Db/fermchop.db (hunk 2 above)
Comment (FreddieAkeroyd) [reply FERMCHOP/FERMCHOP-IOC-01App/Db/fermchop.db]: The PV names are a bit long - also https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/PV-Naming suggests using  TEMP for temperature

**Pull request merged**